### ISSUE TITLE
Tries to close when piping to stdout even though source would not.

### ISSUE THREAD

**Issue by Neppord** [sonewman/cache-stream#1]
I get this when piping cache to stdout. I dont get this message when piping the file directly.

```
events.js:141
      throw er; // Unhandled 'error' event
            ^
Error: process.stdout cannot be closed.
    at WriteStream.stdout.destroy.stdout.destroySoon (node.js:622:20)
    at maybeDestroy (net.js:419:12)
    at WriteStream.Socket.end (net.js:408:5)
    at pump (~/code/test/stream-cache/node_modules/cache-stream/index.js:98:36)
    at doNTCallback0 (node.js:408:9)
    at process._tickDomainCallback (node.js:378:13)
```

**Comment by sonewman** [comment]
hmm, I will take a look. My apologies for the delayed response.

**Comment by sonewman** [comment]
ah this because it tries to call `.end` on `process.stdout` which you cannot do.

Thanks for letting me know about this bug. That issue is fixed as of a752ff9b641545e651899ec260c8aa147e3149ae

It's worth noting that after the initial pipe none of the streams events work as they did before.

It has been a long time since i wrote this module, I am thinking it would probably be better to rewrite this module and add an instance method to return a new Stream which pushes through the buffered cache.

That way each consumer will get a new stream instance with fresh stream state.
